pong.cpp

```diff
@@ -20,15 +22,110 @@ void Pong::PongInit()
     auto ball = std::make_unique<Ball>();
     ball->OnStart();
     mGameObjects.push_back(std::move(ball));
+
+    const float horizontalWallWidth = 640 * 2;
+    const float horizontalWallHeight = 25;
+    const float horizontalWallY = 480;
+    const float verticalWallWidth = 25;
+    const float verticalWallHeight = 480 * 2;
+    const float verticalWallX = 640;
+
+    auto topWall = std::make_unique<Wall>(horizontalWallWidth, horizontalWallHeight);
+    topWall->OnStart();
+    topWall->SetPosition(glm::vec3(0.0f, horizontalWallY, 0.0f));
```

Comment:
So this pattern repeats itself a handful of times. Does it make sense to have an "Setup" method or similar to a game object that calls OnStart, then positions in it's starting position? Saves yourself from some repetition. You could also consider OnStart setting the initial position

If you wanted to be very extra you could make a factory method for these which would do the pattern:

```
obj = make_unique(args)
obj.SetUp()
```

Reply:
Maybe. I think it might be alright for now. I have been thinking about how this kind of setup would work. Modelling things from unity, you would place all your gameobjects in the world in the 3D viewer for however you want the game to start. You would set their inital positions, names, and even internal variables you chose to expose like speed. So my thinking would be that there are some class specific setup that would be done here that could not really be put into a function.